staging/src/k8s.io/kubectl/pkg/polymorphichelpers/interface.go

```diff
@@ -28,7 +28,7 @@ import (
 )
 
 // LogsForObjectFunc is a function type that can tell you how to get logs for a runtime.object
-type LogsForObjectFunc func(restClientGetter genericclioptions.RESTClientGetter, object, options runtime.Object, timeout time.Duration, allContainers bool) (map[v1.ObjectReference]rest.ResponseWrapper, error)
+type LogsForObjectFunc func(restClientGetter genericclioptions.RESTClientGetter, object, options runtime.Object, timeout time.Duration, allContainers bool, allPods bool) (map[v1.ObjectReference]rest.ResponseWrapper, error)
```

Comment:
I think, instead of changing the signature of this interface (and passing `false` in all the other places), it would be better to define a new `logsForObject` which is specialized by getting logs for all the pods just like [here](https://github.com/kubernetes/kubernetes/blob/e5116a39c96a78511fc8c0da4730d3262b2c121c/staging/src/k8s.io/kubectl/pkg/polymorphichelpers/interface.go#L34)
```go
var AllLogsForObjectFn LogsForObjectFunc = allLogsForObject
```
and a new function, namely `allLogsForObject` in polymorphichelpers. 
Lastly, changing the function to `allLogsForObject`, if all flag is passed in https://github.com/kubernetes/kubernetes/blob/20b28312ea9278c7b93340d0eca71023c45a43b3/staging/src/k8s.io/kubectl/pkg/cmd/logs/logs.go#L265

Because as far as I can see `logsForObject` is used by debug and run commands and they don't seem to be getting logs for **all** pods.